src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
```

Comment:
True, but, we don't always want to use that instance. Using the WorkspaceService.Instance is the default case, but sometimes we want to override the default WorkspaceService -- unit testing for instance. Additionally, we wouldn't use the WorkspaceService if we have a ExecuteStringParams. If we ever introduce a different execute type (executefile, for instance) we wouldn't need the workspace service there either.

I suppose we could have the WorkspaceService be a static property of the ExecuteDocumentSelection and allow it to be reset for testing purposes. What say you to that idea?